### DISCUSSION TITLE
Similarity Search within the dataset

### DISCUSSION THREAD

**Discussion by Vikasdubey0551** [facebookresearch/faiss#3664] [General]
Hello,
I am new to the field of similarity search and have a quick question.
Is there a way/suggestion to obtain buckets containing similar vectors (Let's say with indexLSH) based on some distance criteria within the dataset instead of passing a query vector and doing an exhaustive search?

Thanks!

**Comment by mdouze** [comment]
Do you want to cluster the dataset?

**Comment by Vikasdubey0551** [comment]
@mdouze  Thanks for the reply. Yeah. It definitely sounds like I need some clustering techniques like DBSCAN. However, they are slow and memory intensive so far in my attempts. The speed of faiss and GPU support is really attractive. I am not necessarily looking for extremely accurate results, Just more in line or perhaps a little better than Regular LSH. Any suggestions in that regard?

**Comment by qmpzzpmq** [comment]
Hi, @Vikasdubey0551 did you find anymethod to do with DBSCAN?
